src/libostree/ostree-repo-prune.c

```diff
@@ -160,12 +161,20 @@ _ostree_repo_prune_tmp (OstreeRepo *self,
  * Prune static deltas, if COMMIT is specified then delete static delta files only
  * targeting that commit; otherwise any static delta of non existing commits are
  * deleted.
+ *
+ * This function takes an exclusive lock on the @self repository.
  */
 gboolean
 ostree_repo_prune_static_deltas (OstreeRepo *self, const char *commit,
                                  GCancellable      *cancellable,
                                  GError           **error)
 {
+  g_autoptr(OstreeRepoAutoLock) lock = NULL;
+  lock = ostree_repo_auto_lock_push (self, OSTREE_REPO_LOCK_EXCLUSIVE,
```

Comment:
In new code we usually do declare-and-init.

Reply:
OK. Previously @pwithnall had preferred the lock to be an out parameter, but I kind of like this way. What do you think?

Reply:
Left the interface the way it was and just did it in one line like you say.